src/main/java/org/fenixedu/academic/util/PhoneUtil.java

```diff
@@ -18,134 +18,75 @@
+
+        public boolean shouldReceiveValidationCall(String number);
+
+        public boolean shouldReceiveValidationSMS(String number);
+
+        public boolean isMobileNumber(String number);
 
-    static {
-        FIXED_NUMBERS = new ArrayList<PhoneNumberType>();
-        FIXED_NUMBERS.add(PhoneNumberType.VOIP);
-        FIXED_NUMBERS.add(PhoneNumberType.FIXED_LINE);
+        public boolean isFixedNumber(String number);
+
+        public String getInternationalFormatNumber(String numberText);
+
+        public String resolve(PartyContact contact);
 
-        MOBILE_NUMBERS = new ArrayList<PhoneNumberType>();
-        MOBILE_NUMBERS.add(PhoneNumberType.MOBILE);
     }
 
-    private static final PhoneNumberUtil PHONE_UTIL = PhoneNumberUtil.getInstance();
-    private static final String COUNTRY_CODE = "PT";
+    private static PhoneNumberHandler PHONE_NUMBER_HANDLER = new AcademicPhoneNumberHandler();
 
     public static PhoneNumber getPhoneNumber(String numberText) {
-        if (!StringUtils.isEmpty(numberText)) {
-
-            if (numberText.startsWith("00")) {
-                numberText = numberText.replaceFirst("00", "+");
-            }
-
-            if (isExtension(numberText)) {
-                numberText = getExternalNumberForExtension(numberText);
-            }
-
-            try {
-                final PhoneNumber phoneNumber = PHONE_UTIL.parse(numberText, COUNTRY_CODE);
-                if (PHONE_UTIL.isValidNumber(phoneNumber)) {
-                    return phoneNumber;
-                }
-            } catch (NumberParseException e) {
-                System.out.println("O n�mero n�o � v�lido:" + e);
-                return null;
-            }
-        }
-        return null;
+        return PHONE_NUMBER_HANDLER.parsePhoneNumber(numberText);
     }
 
-    private static PhoneNumberType getPhoneNumberType(PhoneNumber phoneNumber) {
-        return phoneNumber != null ? PHONE_UTIL.getNumberType(phoneNumber) : null;
+    public static String getInternacionalFormatNumber(String numberText) {
+        return PHONE_NUMBER_HANDLER.getInternationalFormatNumber(numberText);
     }
 
     public static boolean isValidNumber(String numberText) {
         final PhoneNumber phoneNumber = getPhoneNumber(numberText);
         return phoneNumber != null;
     }
 
-    private static boolean isType(PhoneNumberType type, Collection<PhoneNumberType> types) {
-        return types.contains(type);
+    public static boolean isMobileNumber(String numberText) {
+        return PHONE_NUMBER_HANDLER.isMobileNumber(numberText);
     }
 
-    public static boolean isMobileNumber(String numberText) {
-        return isType(getPhoneNumberType(getPhoneNumber(numberText)), MOBILE_NUMBERS);
+    public static boolean isFixedNumber(String numberText) {
+        return PHONE_NUMBER_HANDLER.isFixedNumber(numberText);
     }
 
-    public static boolean isPortugueseNumber(String numberText) {
+    @Deprecated
```

Comment:
Why Deprecated? Is it never used? In that case it should be stated.
Is there an alternative method that should be used instead?